### PR TITLE
RMP-12457 Add LogFeeder settings to v2 DL & WL blueprints

### PR DESCRIPTION
**Pull request by lajosrodek**
Property `logfeeder.cloud.storage.mode` gets dynamically generated with the following assumptions:
* DL and standalone WL:
    * `default` (default value), if no cloud storage locations have been defined;
        * Local (implicitly, not checked) Solr only scenario
    * `cloud`, if cloud storage locations have been defined and `INFRA_SOLR` is absent in the current cluster;
        * Cloud storage only scenario
        * Unlikely for DL, but this logic is included anyway
    * `hybrid`, if cloud storage locations have been defined and `INFRA_SOLR` is present in the current cluster
        * Local Solr + cloud storage scenario
* Attached WL:
    * `default` (default value), if no cloud storage locations have been defined;
        * Local (implicitly, not checked) or external Solr only scenario, depending on the presence / absence of `INFRA_SOLR` in the parent DL cluster
    * `cloud`, if cloud storage locations have been defined and `INFRA_SOLR` is absent in the parent DL cluster
        * Cloud storage only scenario
        * No support for local Solr!
    * `hybrid`, if cloud storage locations have been defined and `INFRA_SOLR` is present in the parent DL cluster
        * External Solr + cloud storage scenario

### CODE REVIEW

File: template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+{{{#if-true fileSystemConfigs.cloudStorageEnabled}}}
+  "logfeeder-properties": {
+    "properties": {
+{{{#if-true sharedService.datalakeCluster}}}
+      "logfeeder.cloud.storage.mode": "hybrid"
+{{{else if-true sharedService.attachedCluster}}}
+      "logfeeder.cloud.storage.mode": "hybrid"
+{{{else}}}
+      "logfeeder.cloud.storage.mode": "cloud"
```
Comment (sidseth) [inline template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars]: What's the significance of this? Specific scenario. Standalone cluster with cloud storage will set the mode to "cloud"? What impact will that have.

File: core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp [hunk 2]
```diff
@@ -8,15 +8,24 @@
...
+      {
+        "recovery_settings": []
+      },
+      {
+        "component_settings": []
+      }
     ],
     "configurations": [
+      {
+        "logfeeder-properties": {
```
Comment (sidseth) [inline core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp]: Nit: Could you please move these settings to just before the SmartSense settings. Trying to keep the core services further up in the configs.

Comment (lajosrodek) [reply core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp]: Fixed.

File: core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp [hunk 3]
```diff
@@ -8,17 +8,18 @@
...
+        "recovery_settings": []
+      },
+      {
+        "component_settings": []
+      }
     ],
     "configurations": [
       {
         "core-site": {
+          "fs.defaultFS": "file:///",
```
Comment (lajosrodek) [inline core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp]: @sidseth Added this one as well as requested.

File: template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars (hunk 1 above)
Comment (sidseth) [reply template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars]: That's a good point. The standalone blueprints don't have INFRA_SOLR at the moment.
Explicitly checking logfeeder.cloud.storage.mode makes the most sense to me. That said, this is purely a function of time at this point. The dynamic check for the existence of INFRA_SOLR can happen later.

**Comment by lajosrodek** [comment]
Re-purposing this PR to cover updates to all v2 blueprints: DL, attached WL, standalone WL.

File: template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars (hunk 1 above)
Comment (lajosrodek) [reply template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars]: Fixed, added checking for `INFRA_SOLR` at BP generation time.

**Comment by lajosrodek** [comment]
Having an issue with verifying this in cbd local (2.10.0-dev.377-1-g873150d). Tried creating an attached WL, but it came up non-kerberized despite the secure parent DL. In the absence of Kerberos, property `logsearch-common-env/logsearch_external_solr_kerberos_enabled` is left unset (defaulting to `false`) in the attached WL, so no logs can be sent to the DL Solr this way.

**Comment by lajosrodek** [comment]
@akanto / @doktoric Could you please advise what is wrong with the integration tests?

File: template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplateModelContextBuilder.java [hunk 4]
```diff
@@ -162,4 +162,11 @@ public TemplateModelContextBuilder withFixInputs(Map<String, Object> fixInputs)
         blueprintTemplateModelContext.put(HandleBarModelKey.STACK_VERSION.modelKey(), "{{stack_version}}");
         return blueprintTemplateModelContext;
     }
+
+    private Map<String, Object> createAdjustedFileSystemConfig() {
+        Map<String, Object> result = new HashMap<>();
+        result.putAll(fileSystemConfig);
+        result.put("cloudStorageEnabled", !fileSystemConfig.isEmpty());
```
Comment (sidseth) [inline template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplateModelContextBuilder.java]: This has the potential to byte us for the on-prem use case (Central HDFS, workload clusters do not have an HDFS instance, and are instead configured to point to the central HDFS) or when Ozone comes into play.

That said, I'd imagine the on-prem use case fits very nicely into this model. The name could cause some confusion. Looks good for now.

File: template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars (hunk 1 above)
Comment (lajosrodek) [reply template-manager-blueprint/src/main/resources/blueprints/configurations/logsearch_logfeeder/cloud_storage.handlebars]: @sidseth Please see the updated PR description above if you are happy with the current logic. (Especially the attached WL case, where no support for a local Solr is given at the moment.)

File: core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp (hunk 3 above)
Comment (lajosrodek) [reply core/src/main/resources/defaults/blueprints/hdp30-shared-services-v2.bp]: Moved to #4355 for urgency.

**Comment by lajosrodek** [comment]
Has been rebased onto `master` 8893441b5d393c77796925602b110b89215251b8.

**Pull request merged**